app/Policies/UserGameListEntryPolicy.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Policies;
+
+use App\Models\Role;
+use App\Models\User;
+use App\Models\UserGameListEntry;
+use Illuminate\Auth\Access\HandlesAuthorization;
+use Illuminate\Auth\Access\Response;
+
+class UserGameListEntryPolicy
+{
+    use HandlesAuthorization;
+
+    public function viewAny(?User $user): bool
+    {
+        if (!$user) {
+            return false;
+        }
+
+        if ($user->hasAnyRole([
+            Role::MODERATOR,
+            Role::ADMINISTRATOR,
+        ])) {
+            return true;
+        }
+
+        return false;
+    }
+
+    // public function view(User $user, UserGameListEntry $userGameListEntry): bool
+    // {
+    //     return $user->ID === $userGameListEntry->user_id
+    //         ? Response::allow()
+    //         : Response::denyWithStatus(401);
+    // }
+
+    public function view(User $user, User $targetUser): Response
+    {
+        return ($user->id === $targetUser->id) || $user->isFriendsWith($targetUser)
+        ? Response::allow()
+        : Response::denyWithStatus(401);
+    }
```

Comment:
I wanted to have a way to override the `view(User, UserGameListEntry)` method, but I was getting yelled at by IDE for having two of the same method names. Any recommendations on this?